src/LibraryPage.tsx

```diff
@@ -114,6 +115,29 @@ const LibraryPage: React.FC = () => {
     },
     [books]
   );
+  const editBookDatabase = useCallback((book: Book) => {
+    let BookString = JSON.stringify(book);
+    fetch(`http://paralibrary.digital/api/books/${book.id}`, {
+      method: "PUT",
+      credentials: "include",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: BookString,
+    })
+      .then((res) => {
+        if (res.ok) {
+          let newBooks = books.map((b) => {
+            if (b.id === book.id) {
+              return book;
+            }
+            return b;
+          });
+          setBooks(newBooks);
```

Comment:
Does this work now? Or are you still running into the issue where `books` is empty?

Reply:
I am still running into the same issue

Reply:
Oh, it's a missing dependency in the callback. That means that the function wasn't being recreated when the books state changes and since books starts as an empty array it would always be empty.
`useCallback(function, [deps])` creates a function that only recomputes when something in the list `deps` changes.
Always check the linter warnings.
<img width="512" alt="Screenshot 2020-05-26 17 17 39" src="https://user-images.githubusercontent.com/33438160/82962366-df52f500-9f74-11ea-99ad-5c5f8ce18dad.png">